### PR TITLE
Fix Sirens not removed correctly 

### LINKED ISSUE #3472: addVehicleSirens not working correctly
### Describe the bug

addVehicleSirens when readd the Sirens to same vehicle it will not show up

### Steps to reproduce
addSirens
removeSirens
addSirens agian to same car bug happend
[https://youtu.be/1AVyNDfCKG8](https://youtu.be/1AVyNDfCKG8)

### Version

Multi Theft Auto v1.6-release-22518

### Additional context

```
addEventHandler('onPlayerVehicleEnter', root, function(vehicle, seat)
  addVehicleSirens(vehicle, 7, 2, false,  false, false,  true)
      setVehicleSirens(vehicle, 1, -0.3, 0, 0.8, 0, 0, 255, 255, 255)
      setVehicleSirens(vehicle, 2, 0.3, 0, 0.8, 255, 0, 0, 255, 255)
	   setVehicleSirensOn(vehicle, true)

end)

addEventHandler('onPlayerVehicleExit', root, function(vehicle, seat)
   removeVehicleSirens(vehicle)
end)
```

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not security related.

### PR DESCRIPTION
**Pull request by Proxy-99**
attempt to fix #3472 

the issue with SetSirenActive() it return true after removing the sirens 
deactive the sirens when removing them

### CODE REVIEW

**Comment by Dutchman101** [comment]
> attempt to fix #3472

Attempt? It should be straightforward for you to test your changes (against the issue repro), please do so.

**Comment by Proxy-99** [comment]
> > attempt to fix #3472
> 
> Attempt? It should be straightforward for you to test your changes (against the issue repro), please do so.

tested issue gone

**Comment by Proxy-99** [comment]
> > attempt to fix #3472
> 
> Attempt? It should be straightforward for you to test your changes (against the issue repro), please do so.

the issue with `SetSirenActive() `it return true after removing the sirens 
which makes the function return false in the second time you fire the SetVehicleSirensOn with same vehicle

```
bool CStaticFunctionDefinitions::SetVehicleSirensOn(CElement* pElement, bool bSirensOn)
{
    assert(pElement);
    RUN_CHILDREN(SetVehicleSirensOn(*iter, bSirensOn))

    if (IS_VEHICLE(pElement))
    {
        CVehicle* pVehicle = static_cast<CVehicle*>(pElement);

        // Has Sirens and different state than before?
        if ((CVehicleManager::HasSirens(pVehicle->GetModel()) || pVehicle->DoesVehicleHaveSirens()) && bSirensOn != pVehicle->IsSirenActive())
        {
            // Set the new state
            pVehicle->SetSirenActive(bSirensOn);

            // Tell everyone
            unsigned char ucSirensOn = 0;
            if (bSirensOn)
                ucSirensOn = 1;

            CBitStream BitStream;
            BitStream.pBitStream->Write(ucSirensOn);
            m_pPlayerManager->BroadcastOnlyJoined(CElementRPCPacket(pVehicle, SET_VEHICLE_SIRENE_ON, *BitStream.pBitStream));
            return true;
        }
    }

    return false;
}
```


`bSirensOn != pVehicle->IsSirenActive()` will return false in the second time you fire the function with same car

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 1]
```diff
@@ -5008,6 +5008,7 @@ bool CStaticFunctionDefinitions::RemoveVehicleSirens(CVehicle* pVehicle)
 {
     assert(pVehicle);
 
+    pVehicle->SetSirenActive(false);
```
Comment (tederis) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: Please, move this line under the next line.

**Review by tederis — APPROVED** [review]

**Comment by Dutchman101** [comment]
Thanks!
@Proxy-99

**Pull request merged**